### PR TITLE
feat: initialize unsloth skill module with comprehensive documentation

### PR DESCRIPTION
**Pull request by Hyperakan**

### CODE REVIEW

File: assets/onboarding/skills/unsloth/tests/test_skill.py
Comment (monatis) [inline assets/onboarding/skills/unsloth/tests/test_skill.py:1]: This file shouldn't be a part of the skill.

File: assets/onboarding/skills/unsloth/references/dataset_and_chat_templates.md
Comment (monatis) [inline assets/onboarding/skills/unsloth/references/dataset_and_chat_templates.md:1]: `get_chat_template()` is confusing and unnecessary for most modern models. Historically, some models didn't ship a chat template defined in their tokenizer, prior to HuggingFace standardizing a format for this. `get_chat_template()` was a helper to set a template by name. Now, almost all the models define their own chat template correctly and thus no need for this patch for them. Agents (and of course, humans as well) should be aware of and careful about this. `get_chat_template()` should be preferred only when the model does not have a chat template defined.

File: assets/onboarding/skills/unsloth/references/sequence_packing_and_prefix_caching.md
Comment (monatis) [inline assets/onboarding/skills/unsloth/references/sequence_packing_and_prefix_caching.md:1]: To my knowledge, only with Flash Attention does sequence packing work correctly. All other attention kernels apply an incorrect attention mask for sequence packing. Additionally, it usually has a tradeoff, i.e., throughput increase at the cost of a slight degredation in trained model performance, especially in multi-epoch runs, because it hinders randomization across epochs. Agents and humans alike should be are of these gotchas.

File: assets/onboarding/skills/unsloth/references/triton_kernels_and_architecture_internals.md
Comment (monatis) [inline assets/onboarding/skills/unsloth/references/triton_kernels_and_architecture_internals.md:1]: I'm not sure if we really need this.

File: assets/onboarding/skills/unsloth/scripts/benchmark_memory_speed.py
Comment (monatis) [inline assets/onboarding/skills/unsloth/scripts/benchmark_memory_speed.py:1]: As we did in other skills, we need an Inline Script Metadata block according to the PEP 723 standard. I saw one or more files already have it, but we need to standardize this behavior. Additionally, we better have those dependencies pinned to constrained and compatible versions, because these packages can evolve quickly. In addition to the `test_skill` script, which should be moved out of the skill to a separate `scripts` directory, we can also have a utility script to bump these versions for maintenance when needed.

File: assets/onboarding/skills/unsloth/scripts/finetune_sft.py
Comment (monatis) [inline assets/onboarding/skills/unsloth/scripts/finetune_sft.py:1]: `qwen 2.5` is too old to be the default. LLMs already have a tendency to blindly use an old model variant due to their knowledge cutoff dates. `isanagent` has a specific instruction in its prompt to tell the agent to avoid relying on their internal knowledge for versions and try to obtain up-to-date information online. Such old reference may hinder this behavior. Instead, we need to set up-to-date defaults as much s possible and still tel the agent this information may be obsoleted and check the current version and use it if any unless explicitly stated otherwise. This comment also applies to some other files where we have model names defined. Safer defaults may include qwen3.5 or gemma4 and qwen3.6 for slightly larger models (qwen3.6's smallest variant is 27b).

File: assets/onboarding/skills/unsloth/SKILL.md
Comment (monatis) [inline assets/onboarding/skills/unsloth/SKILL.md:1]: This `SKILL.md` contains too much self-promotion nonsense by Unsloth such as 80% less memory, 1.6x speedup etc. which are measured against naive baselines and hype words that we need to avoid including here. The description and the content should be modified accordingly. Additionally, it contains very long code snippets. We need to remove them and instead refer to individual reference and script files. This file should be like "table of contents" and "preface," e.g., general guidelines that should be applied in almost all cases such as running scripts with `--help` flag first to learn what arguments it accepts, copy and modify the script if they want to adjust it to somewhere else or do something differently, be aware of Inline Script Metadata defined in scripts, check model names and use a recent model variant unless a specific model name is explicitly stated, measure the performance and compare it with the baseline etc. plus paths and descriptions of other files in the skill.

**Review by monatis — CHANGES_REQUESTED** [review]
This is a good start, but some issues need to be addressed before merging this PR.

File: assets/onboarding/skills/unsloth/scripts/benchmark_memory_speed.py
Comment (monatis) [inline assets/onboarding/skills/unsloth/scripts/benchmark_memory_speed.py:1]: All these package versions seem to be too old --from at least one year ago or something.

File: assets/onboarding/skills/unsloth/SKILL.md
Comment (monatis) [inline assets/onboarding/skills/unsloth/SKILL.md:1]: Instead of giving exact commands such as "Run `python3 scripts/audit_dataset.py`" etc., prefer statements such as "Run `scripts/audit_dataset.py`" etc. because the agent will need to use different commands: `python3` only with system-wide installations on some operating systems, `python` for most other environments, `uv` in more modern environments, and `cinderflow exec ...` in most of the cases with isanagent. The agent will eventually discover it, but sometimes, especially with lower-tier models, it just gets confused and wastes time on it.

**Review by monatis — CHANGES_REQUESTED** [review]
Great improvemeent over the initial PR. We only need slight polishing in `SKILL.md` and probably version bumps in scripts because they seem to be a little bit outdated. There might be a need for slight API changes in scripts after bump versions.

**Review by monatis — APPROVED** [review]
Great, thank you.
